pytamaro/graphic.py

```diff
@@ -242,15 +274,78 @@ class Rotate(Graphic):
     """
     Represents the rotation of a graphic by a certain angle clockwise.
     """
+
     def __init__(self, graphic: Graphic, deg: float):
         self.graphic = graphic
         self.rot_matrix = Matrix.RotateDeg(deg, self.graphic.pin_position)
         self.path = Path()
         self.graphic.path.transform(self.rot_matrix, self.path)  # updates self.path
         self.pin_position = graphic.pin_position
+        # For graphic tree comparison
+        self.degree = - deg
 
     def draw(self, canvas: Canvas):
         canvas.save()
         canvas.concat(self.rot_matrix)
         self.graphic.draw(canvas)
         canvas.restore()
+
+
+class Beside(Graphic):
+    """
+    Represent the composition of two graphics that beside with each other,
+    the center of the two graphics are on the same horizontal level.
+    """
+
+    def __init__(self, left_graphic: Graphic, right_graphic: Graphic):
+        self.left_graphic = left_graphic
+        self.right_graphic = right_graphic
+
+        b_g = Pin(Compose(Pin(left_graphic, center_right), Pin(right_graphic, center_left)), center)
+
+        self.path = Path(b_g.path)
+        self.set_pin_position(b_g.pin_position.x(), b_g.pin_position.y())
+
+    def draw(self, canvas: Canvas):
+        b_g = Pin(Compose(Pin(self.left_graphic, center_right), Pin(self.right_graphic, center_left)), center)
```

Comment:
Oh no, let's not compute this twice (and also duplicate code)!
It can be stored as an attribute.